### ISSUE TITLE
Trapping connection errors.

### ISSUE THREAD

**Issue by bplatz** [neumino/thinky#210]
I haven't been able to find a good way to trap connection errors. I realize that max retries and other options can be set when creating the connection but a try / catch doesn't seem to get the error once the max retries has expired.

It seems an event emitter on the connection would be ideal, the only event emitters (at least documented) are on models and documents.

What I need is a way to capture when this happens to send off alerts.

**Comment by neumino** [comment]
Thinky uses rethinkdbdash and does not access to any connections.
Connection errors are bubbled up to queries when you run them.

You can access rethinkdbdash with `thinky.r` and then access the pool with `thinky.r.getPool()` if you want, but network errors should be properly handled by the driver.

**Comment by neumino** [comment]
Rethinkdbdash 2.1 will come with a way to trap those errors. Once it released (with thinky 2.1), the syntax will be:

```
var thinky = require('thinky')();
thinky.r.getPoolMaster().on('log', console.log)
```
